flink-table/flink-table-planner/src/main/java/org/apache/calcite/schema/Statistic.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.schema;
+
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelDistribution;
+import org.apache.calcite.rel.RelReferentialConstraint;
+import org.apache.calcite.util.ImmutableBitSet;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Statistics about a {@link Table}.
+ *
+ * <p>Each of the methods may return {@code null} meaning "not known".</p>
+ *
+ * <p>Changes:
+ *
+ * <ul>
+ *     <li>Line 61: default collations change from null to empty list.</li>
```

Comment:
Is it a temporary solution? Why we have to override the default value in interface?
But this change is only needed in legacy planner?

Reply:
Because the default value changes to null, the legacy planner does not have custom `Statistic` so there is no good way to override the default value, e.g. (the `Statistics.UNKNOWN`).

Reply:
Yes, but because the legacy planner use the `RelOptTableImpl` as the `SqlValidatorTable` during sql-to-rel conversion, and there is no common base preparing table like `FlinkPreparingTableBase` in blink planner, we need to override each `AbstractTable` sub-class for the legacy planner.

Here it is `QueryOperationCatalogViewTable` that causes the problem.